### PR TITLE
Add generic APPUiO Managed OpenShift 4 architecture page

### PR DESCRIPTION
**Pull request by simu**

### CODE REVIEW

File: docs/modules/ROOT/pages/references/architecture/index.adoc [hunk 1]
```diff
@@ -0,0 +1,34 @@
+= APPUiO Managed OpenShift 4 Architecture
+
```
Comment (corvus-ch) [inline docs/modules/ROOT/pages/references/architecture/index.adoc:2]: This is quite "in medias res" and might catch potential readers on the wrong foot. Please add some introductory text.

* State that a VSHN Managed OpenShift cluster builds on top of Red Rat OpenShift which builds on top of Kubernetes
* Explain that general knowledge about such clusters is assumed and that only things specific to VSHN are mentioned here
* Provide links where one can read up on that general knowledge. Some ideas:
    * https://access.redhat.com/documentation/en-us/openshift_container_platform/4.14/html/architecture/index
    * https://docs.openshift.com/container-platform/latest/architecture/index.html
    * https://developer.ibm.com/blogs/openshift-101-architecture/
    * https://www.redhat.com/en/topics/containers/kubernetes-architecture
    * https://kubernetes.io/docs/concepts/architecture/
    * https://devopscube.com/kubernetes-architecture-explained/

File: docs/modules/ROOT/pages/references/architecture/index.adoc [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+Every APPUiO Managed OpenShift 4 has 3 control plane nodes.
+Additionally, APPUiO Managed OpenShift 4 clusters have https://docs.openshift.com/container-platform/4.13/nodes/nodes/nodes-nodes-creating-infrastructure-nodes.html[4 infrastructure nodes].
+Finally, APPUiO Managed OpenShift 4 clusters have 3 worker nodes by default.
+The minimum size for each node is 4 vCPUs, 16 GiB of RAM and 100 GiB root disk.
```
Comment (corvus-ch) [inline docs/modules/ROOT/pages/references/architecture/index.adoc:6]: Omit adding specific values here. Instead, state that there are minimum requirements. In the infra specific architectures, we then should link to https://products.vshn.ch/appuio/managed/ocp4_cloudscale.html#_default_configuration_minimum_requirements.

File: docs/modules/ROOT/assets/images/ocp4-external-dependencies.svg
Comment (corvus-ch) [inline docs/modules/ROOT/assets/images/ocp4-external-dependencies.svg:1]: Can we make it clear, that OpsGenie and Passbolt are required for VSHN to fulfill our operational responsibility?

File: docs/modules/ROOT/assets/images/ocp4-architecture.svg
Comment (corvus-ch) [inline docs/modules/ROOT/assets/images/ocp4-architecture.svg:1]: Would it make sense, to have a perspective that is one level deeper? Showcase the namespace separation and hint NetworkPolicies as a tool to control who is allowed to connect with whom?

Another perspective is what services are available on all clusters and who needs to access them:
* VSHNeers: API and Console
* Customer Developers: API and Console
* Application users: Application

**Review by corvus-ch — CHANGES_REQUESTED** [review]
Please add:

* an introductory text with links to further reading sources.
* a perspective of who needs to access what on the cluster and their network context they are cumming from (especially VSHN).

File: docs/modules/ROOT/pages/references/architecture/index.adoc (hunk 2 above)
Comment (simu) [reply docs/modules/ROOT/pages/references/architecture/index.adoc]: Adjusted to reflect the actual minimum requirements of Red Hat OpenShift 4. I looked through our variants, and they all have the same minimum requirements, so I left our base line configuration here, but I can remove it if that works better for you.

File: docs/modules/ROOT/assets/images/ocp4-architecture.svg
Comment (simu) [reply docs/modules/ROOT/assets/images/ocp4-architecture.svg:1]: I don't think the namespace granularity perspective is necessary (or desired) in the architecture documentation. If we want to have a namespace granularity level documentation, it should be a separate page.

Comment (simu) [reply docs/modules/ROOT/assets/images/ocp4-architecture.svg:1]: I updated the diagram and added a section on different user access patterns.

File: docs/modules/ROOT/pages/references/architecture/index.adoc [hunk 3]
```diff
@@ -0,0 +1,78 @@
+= APPUiO Managed OpenShift 4 Architecture
+
+APPUiO Managed OpenShift 4 is based on https://docs.openshift.com/container-platform/latest/welcome/index.html[Red Hat OpenShift 4 Container Platform], which is a https://kubernetes.io[Kubernetes] distribution maintained and developed by Red Hat.
+This documentation assumes that readers are familiar with Kubernetes concepts.
+
+For details on the architecture of a Kubernetes cluster, we recommend reading the https://kubernetes.io/docs/concepts/architecture/[cluster architecture section] of the official Kubernetes documentation.
+
+For details on the architecture of a Red Hat OpenShift 4 cluster, please see the https://access.redhat.com/documentation/en-us/openshift_container_platform/4.14/html/architecture/index[architecture section] of the OpenShift 4 product documentation or https://docs.openshift.com/container-platform/latest/architecture/index.html[architecture page] in the OpenShift Container Platform technical documentation.
```
Comment (corvus-ch) [inline docs/modules/ROOT/pages/references/architecture/index.adoc:8]:
```suggestion
For details on the architecture of a Red Hat OpenShift 4 cluster, please see the https://access.redhat.com/documentation/en-us/openshift_container_platform/latest/html/architecture/index[architecture section] of the OpenShift 4 product documentation or https://docs.openshift.com/container-platform/latest/architecture/index.html[architecture page] in the OpenShift Container Platform technical documentation.
```

File: docs/modules/ROOT/pages/references/architecture/index.adoc [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+* cloudscale.ch (coming soon)
+* Exoscale (coming soon)
+* VMWare vSphere (coming soon)
+
+Additionally, the https://products.vshn.ch/appuio/managed/ocp4.html[APPUiO Managed OpenShift 4 product documentation] provides more details on the minimum required resources, supported Red Hat OpenShift 4 features and components as well as optionally supported features and add-ons for each supported infrastructure provider.
+
+== References
+
+* https://access.redhat.com/documentation/en-us/openshift_container_platform/4.14/html/architecture/index[Red Hat OpenShift 4 architecture]
```
Comment (corvus-ch) [inline docs/modules/ROOT/pages/references/architecture/index.adoc:77]:
```suggestion
* https://access.redhat.com/documentation/en-us/openshift_container_platform/latest/html/architecture/index[Red Hat OpenShift 4 architecture]
```

**Review by corvus-ch — APPROVED** [review]
Use latest instead of a specific version when linking to the OCP documentation.

**Pull request merged**